Kernel/Net/Realtek/RTL8168NetworkAdapter.cpp

```diff
@@ -931,6 +1043,49 @@ void RTL8168NetworkAdapter::hardware_quirks_b_2()
     out8(REG_CONFIG4, in8(REG_CONFIG4) & ~1);
 }
 
+void RTL8168NetworkAdapter::hardware_quirks_c_1()
+{
+    csi_enable(CSI_ACCESS_2);
+
+    out8(REG_DBG, 0x06 | DBG_FIX_NAK_1 | DBG_FIX_NAK_2);
+
+    constexpr EPhyUpdate ephy_info[] = {
+        { 0x02, 0x0800, 0x1000 },
+        { 0x03, 0, 0x0002 },
+        { 0x06, 0x0080, 0x0000 }
+    };
+    extended_phy_initialize(ephy_info, 3);
```

Comment:
All these magic numbers do not spark joy. For the quirks functions, and the configure functions.  Is there really no reference online that explains what the values mean? No manual?

Reply:
I'm also not satisfied with those black-box values. I have looked through various websites and I didn't find anything meaningful :(

Reply:
There's no manual for most of the RTL8168 chips out there. Therefore the Linux driver has these black-box values and we just have to trust them... or maybe not support these devices at all?
I mean, we have Intel e1000 and e1000e devices - I even bought one (it was about 300 ILS, about 80 USD) for testing with bare metal serenity setup, and it worked quite OK, so it's not like we don't have better options (and Intel is much better imho than Realtek for Ethernet devices).

Frankly, I don't see a good value in adding quirks for such devices when we simply don't know what's wrong with them - this is not about being a free-and-open-source software zealot, it's about the fact that we don't have an idea how to fix bugs with these devices, should we ever encounter a problem with them.
As far as I know and remember, we never had black-box magic values for any driver in the kernel, and we might want to keep it this way so we keep the kernel code understandable for everyone.

Does your device not work correctly at all without these values?

Reply:
> or maybe not support these devices at all?

In my opining having better hardware support is a nice thing to have, considering that the work is already done. Few black-box values don't make entire codebase unreadable. Unfortunately this is how it is with commercial hardware.

> Does your device not work correctly at all without these values?

It seems to work the same way without writing these values but I have no idea how reliable that is. I dual boot SerenityOS and Linux Mint on that PC so perhaps it works just because Linux driver wrote PHY registers and handled hw quirks before and they were kept.